File: Patrones Comportamiento/State/State.js
// State
class State {
  handle() {}
}

class Active extends State {
  handle() {
    console.log("Estado activo");
  }
}

class Context {
  setState(state) {
    this.state = state;
  }
  request() {
    this.state.handle();
  }
}

const c = new Context();
c.setState(new Active());
c.request();

File: Patrones Comportamiento/State/State.py
# State
class State:
    def handle(self): pass

class Active(State):
    def handle(self):
        print("Estado activo")

class Context:
    def set_state(self, state):
        self.state = state

    def request(self):
        self.state.handle()

c = Context()
c.set_state(Active())
c.request()
